scale-tests/batch_test.py

```diff
@@ -85,14 +90,14 @@ def submit_job(dispatcher: typing.Dict, duration: int, config: typing.List[str])
     else:
         spark_utils.submit_job(
             service_name=dispatcher_name,
-            app_url=MONTE_CARLO_APP_URL,
+            app_url=app_url,
             app_args=app_args,
             verbose=False,
             args=config,
             driver_role=dispatcher["roles"]["executors"])
```

Comment:
I know it wasn't done in this PR, but could we use local variables for `spark_user` and `principal` and set these based on the strict mode. instead of having two calls to the same function.